### DISCUSSION TITLE
Why does ` CmaEsSampler` expect at least 2D continuous space?

### DISCUSSION THREAD

**Discussion by ego-thales** [optuna/optuna#5524] [Q&A] [answered]
### Motivation

I think it seems reasonable to sample 1D parameter space according to CMAES strategy, so I wonder why it doesn't work. Maybe this edge case overlaps with `GPSampler` but I think there's more to it (population evolution etc...). Am I missing something?

### Description

When trying to use `CmaEsSampler` on 1D parameter space, we get the following warning:

> `CmaEsSampler` only supports two or more dimensional continuous search space. `RandomSampler` is used instead of `CmaEsSampler`

### Alternatives (optional)

Make it compatible with 1D parameter space, to avoid creating a dummy variable.

**Comment by y0z** [comment]
Thank you for your question.

This restriction comes from the cmaes library.
https://github.com/CyberAgentAILab/cmaes/blob/main/cmaes/_cma.py#L90

Therefore, I think once the cmaes library supports 1D search space, Optuna could also support it.

**Comment by y0z** [comment, marked as answer]
FYI, this problem was resolved in v4.5.0 (see https://github.com/optuna/optuna/releases/tag/v4.5.0).
